### ISSUE TITLE
Detect event stopJob

### ISSUE THREAD

**Issue by raabbajam** [Narzerus/yakuza#53]
Currently I wait on these two events

``` js
   job.on('job:fail', function (response) {
      // console.log('Job failed', response.error);
      reject(new Error(response.error));
    });
    job.on('task:getAccountData:success', function (response) {
      resolve(response.data);
    });
```

But when preceding task (tasks before `getAccountData`) succeed with undesired result (hence I do `task.stopJob`), neither of them event raised.
Is there any way to detect event stopJob?

**Comment by RafaelVidaurre** [comment]
Hmm, can't remember for sure if there is an event for that. 

So, clearly since the task previous to `getAccountData` stops the execution chain, those events you are listening to are not triggereed (because the task never started)

Why not trigger a `task.fail()` in the previous task and listen to that?

Rhere is also an event called `job:finish` which might or might not be of use  depending on what you are trying to achieve

**Comment by raabbajam** [comment]
From my understanding based on the concept, `task.fail` should be used only on an error request process. Undesired result is a success, since it still considered successful request. And actually if we separate request process and data validation process concerns, this concept does perfectly make sense.

`hooks`, `stopJob` are nice, but if we don't have an event for that, how do we know when it is stopped. These features provide result validation, and `return early` way on _succesful yet undesired outcome_.

Another sample case is real time search, when query result not found, just return early, but we will not call it failed, instead we usually return `succes: true, data: empty array`.

What do you think?

**Comment by RafaelVidaurre** [comment]
Yeah, I agree, `fail` should only be used if something went wrong. If _undesired outcome_ is an expected result then success should be used.

Can you submit a **feature request** for this? I cannot implement this right away so in the meantime a feature request can bing in a PR. You can submit a PR too if you feel like it!
